firebase/firebase-tools
# Allow specifying the region for Hosting functions rewrites, defaulting to us-central1 if not provided

### Description

In the firebase.json > hosting > rewrites, add a field `region` to specify that a function is in a non-`us-central1` region.  Does nothing if `function` is not also specified.  If the region is not specified, default to `us-central1`.

### Scenarios Tested

Set the hosting config to the following
```
  "hosting": {
    "public": "public",
    "rewrites": [
      {
        "source": "/testing",
        "function": "testing",
        "region":"europe-west1"
      }
    ]
  }
``` 
and create a function:
```
exports.testing = functions
.region('europe-west1', 'europe-west2')
.https.onRequest(
  async (request, res) => {

    res.send(`target: ${process.env.FUNCTION_TARGET}`)
  },
);
```

- Run the emulator, and the function will properly get rewritten to. 
- Change `europe-west1` to `us-central1`, and the rewrite will no longer work.

Also, deploy hosting with the `--debug` flag, and notice the uploaded config includes the region:
```
 {"config":{"rewrites":[{"glob":"/testing","function":"testing","function_region":"europe-west1"}]},"labels":{"deployment-tool":"cli-firebase"}}
 ```
